### PR TITLE
Token Expired Updates

### PR DESCRIPTION
**Pull request by MatkoTom**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant
motivation and context. List any dependencies that are required for this change.

Updated logic to handle token expiration. Users will no longer be able to enter app if token expired. Instead, they will be sent to the Onboarding screen with their number locked in, until they log in again.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration. All testing was done
manually by following, plainly, functionality requirements. If there are some special edge cases or
special ways that things need to be tested they will be mentioned below.

Dev testing

**Test Configuration**:

* Firmware version: G980FXXSFFVHA
* Hardware: SAMSUNG Galaxy S20
* SDK: Android 13

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/src/main/java/com/clover/studio/spikamessenger/BaseViewModel.kt [hunk 1]
```diff
@@ -2,11 +2,19 @@ package com.clover.studio.spikamessenger
...
 import com.clover.studio.spikamessenger.utils.helpers.Resource
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 import timber.log.Timber
 
 open class BaseViewModel : ViewModel() {
+
+    @Inject
+    lateinit var sharedPrefs: SharedPreferencesRepository
+
```
Comment (ivolarevic) [inline app/src/main/java/com/clover/studio/spikamessenger/BaseViewModel.kt:17]: Why empty line here?

File: app/src/main/java/com/clover/studio/spikamessenger/ui/splash/SplashViewModel.kt [hunk 2]
```diff
@@ -17,14 +17,14 @@ class SplashViewModel @Inject constructor(
     var splashTokenListener = MutableLiveData<Event<SplashStates>>()
 
     fun checkToken() = viewModelScope.launch {
-        if ((sharedPrefsRepo.readToken()
-                .isNullOrEmpty() && !sharedPrefsRepo.isAccountCreated()) || !sharedPrefsRepo.readRegistered()
-        ) {
-            splashTokenListener.postValue(Event(SplashStates.NAVIGATE_ONBOARDING))
-        } else if (!sharedPrefsRepo.readToken().isNullOrEmpty()
+        if (!sharedPrefsRepo.readToken().isNullOrEmpty()
```
Comment (ivolarevic) [inline app/src/main/java/com/clover/studio/spikamessenger/ui/splash/SplashViewModel.kt:20]: Are you sure that is ok to use isNullOrEmpty and not == ""? I remember that we were looking at that method which sometimes does not recognize that it is an empty string even though it was?

File: app/src/main/java/com/clover/studio/spikamessenger/ui/splash/SplashViewModel.kt [hunk 3]
```diff
@@ -17,14 +17,14 @@ class SplashViewModel @Inject constructor(
...
             && !sharedPrefsRepo.isAccountCreated()
         ) {
             splashTokenListener.postValue(Event(SplashStates.NAVIGATE_ACCOUNT_CREATION))
+        } else if (sharedPrefsRepo.readToken()
+                .isNullOrEmpty() || !sharedPrefsRepo.readRegistered()
```
Comment (ivolarevic) [inline app/src/main/java/com/clover/studio/spikamessenger/ui/splash/SplashViewModel.kt:25]: Same as above

File: app/src/main/java/com/clover/studio/spikamessenger/ui/splash/SplashViewModel.kt (hunk 2 above)
Comment (MatkoTom) [reply app/src/main/java/com/clover/studio/spikamessenger/ui/splash/SplashViewModel.kt:20]: Wasn't the issue that the .toString() method turned null fields into ""? Thus, when checking for null it would pass as ok, especially nullable fields in that case, contained in objects.

**Review by ivolarevic — APPROVED** [review]
